### ISSUE TITLE
[Diff view] As a user, I would like to see the files in the same order as in GitHub

### ISSUE THREAD

**Issue by kittaakos** [gitpod-io/gitpod#298]
For instance, the order of the changes in GH is:
```
 .travis.yml
 examples/browser/package.json
 packages/git/package.json
 packages/git/src/browser/git-view-contribution.ts
 packages/plugin-ext/package.json
...
```

In Gitpod/Theia:
![screen shot 2019-02-08 at 13 33 27](https://user-images.githubusercontent.com/1405703/52478711-8821e900-2ba6-11e9-9587-896531ac03ef.jpg)

The PR I have used: https://github.com/theia-ide/theia/pull/4279/files

**Comment by svenefftinge** [comment]
Just to better understand. Is it so they are aligned, or is there something that is better about the ordering on GitHub?

**Comment by kittaakos** [comment]
> aligned

It is about alignment.

**Comment by kittaakos** [comment]
TL;DR: do no mimic GH's ordering but pimp up ours.

> > aligned
> 
> It is about alignment.

It's a bad idea. What if GitHub changes the ordering let's say in a year, or GitLab uses a different ordering strategy than GH? Yes, we can support different strategies in Gitpod, but perhaps it does not worth it. We should stick to a consistent GP user experience no matter the user is browsing a pull request on GH or GL.

But, it would do make sense to be a bit smarter when ordering the change items. Instead of using the name only, we should consider the path too. For instance, having the `tsconfig.json` and the `.travis.yml` would be nicer; they are on the same level.

As a pipe dream, we can support multiple built-in ordering rules, and the user can pick one from the preferences. Does this worth a ticket in Theia? I guess this is not a GP issue.

**Comment by svenefftinge** [comment]
Have you looked into how the ordering on GitHub works? I think it would be good to align. I don't care about future possible problems as much as about the current experience :-)
As a user coming from a GH PR and being used to that ordering it would be great to have the same in Gitpod. In case GL has different ordering we could use a different sorting strategy. There is not an endless amount of possible sorting strategies here so that would be fine.

**Comment by svenefftinge** [comment]
> Instead of using the name only, we should consider the path too.

Seems like this is exactly the thing that GitHub does differently, no?

**Comment by kittaakos** [comment]
> Seems like this is exactly the thing that GitHub does differently, no?

Yes. I have checked it once more, and it seems, GH compares the relative path segments. Interestingly, `tsconfig.json` is **not** next to `.travis.yml`. The order from GH:

```
.travis.yml
examples/browser/package.json
packages/git/package.json
packages/git/src/browser/git-view-contribution.ts
packages/plugin-ext/package.json
packages/plugin-ext/src/api/plugin-api.ts
packages/plugin-ext/src/main/browser/main-context.ts
packages/plugin-ext/src/main/browser/scm-main.ts
packages/plugin-ext/src/plugin/plugin-context.ts
packages/plugin-ext/src/plugin/scm.ts
packages/scm/src/browser/index.ts
packages/scm/src/browser/scm-contribution.ts
packages/scm/src/browser/scm-frontend-module.ts
packages/scm/src/browser/scm-service.ts
tsconfig.json
```

**Comment by svenefftinge** [comment]
Great, I think we should do like that.
